runtime/src/main/cpp/Worker.cpp

```diff
@@ -374,27 +504,46 @@ void* workerRoutine(void* argument) {
     ObjHolder resultHolder;
     while (true) {
       Job job = worker->getJob();
-      if (job.function == nullptr) {
+      if (job.kind == JOB_TERMINATE) {
+        if (job.terminationRequest.waitDelayed) {
+          if (worker->waitDelayed()) {
+            worker->putJob(job, false);
```

Comment:
Is it possible that another job have been added by this moment? This would make worker execute this job despite it is already terminating.

Reply:
Yeah, that's intended behavior. It people want forced termination - it is available.

Reply:
So whether a job added after `requestTermination()` is executed depends on presence of delayed jobs? Is this behaviour intended? Looks a bit confusing.

Reply:
Well, I do not believe it's too problematic. when we start waiting for delayed jobs it's kind of natural that other requests are processed as well. Guess we can fix it if it will indeed become problematic.